File: packages/hmssdk_flutter/ios/Classes/Models/polls/HMSPollLeaderboardResponseExtension.swift
//
//  HMSPollLeaderboardResponseExtension.swift
//  hmssdk_flutter
//
//  Created by Pushpam on 20/02/24.
//

import Foundation
import HMSSDK

class HMSPollLeaderboardResponseExtension {

    static func toDictionary(pollLeaderboardResponse: HMSPollLeaderboardResponse?) -> [String: Any?]? {

        if let pollLeaderboardResponse {

            var map = [String: Any?]()
            var entryMap = [[String: Any?]]()

            for entry in pollLeaderboardResponse.entries {
                entryMap.append(HMSPollLeaderboardEntryExtension.toDictionary(hmsPollLeaderboardEntry: entry))
            }

            map["entries"] = entryMap
            map["has_next"] = pollLeaderboardResponse.hasNext
            map["summary"] = HMSPollLeaderboardSummaryExtension.toDictionary(hmsPollLeaderboardSummary: pollLeaderboardResponse.summary)

            return map
        } else {
            return nil
        }
    }

}

File: packages/hmssdk_flutter/android/src/main/kotlin/live/hms/hmssdk_flutter/poll_extension/HMSPollLeaderboardResponseExtension.kt
package live.hms.hmssdk_flutter.poll_extension

import live.hms.video.polls.network.PollLeaderboardResponse

class HMSPollLeaderboardResponseExtension {
    companion object {
        fun toDictionary(pollLeaderboardResponse: PollLeaderboardResponse): HashMap<String, Any?> {
            val map = HashMap<String, Any?>()

            val entryMap = ArrayList<HashMap<String, Any?>>()

            pollLeaderboardResponse.entries?.forEach { pollLeaderboardResponse ->
                entryMap.add(HMSPollLeaderboardEntryExtension.toDictionary(pollLeaderboardResponse))
            }

            map["entries"] = entryMap
            map["has_next"] = pollLeaderboardResponse.hasNext
            map["summary"] = HMSPollLeaderboardSummaryExtension.toDictionary(pollLeaderboardResponse.summary)
            return map
        }
    }
}
